src/asttypename.d

```diff
@@ -0,0 +1,99 @@
+module ddmd.asttypename;
+
+import ddmd.attrib;
+import ddmd.aliasthis;
+import ddmd.aggregate;
+import ddmd.complex;
+import ddmd.cond;
+import ddmd.ctfeexpr;
+import ddmd.dclass;
+import ddmd.declaration;
+import ddmd.denum;
+import ddmd.dimport;
+import ddmd.declaration;
+import ddmd.dstruct;
+import ddmd.dsymbol;
+import ddmd.dtemplate;
+import ddmd.dversion;
+import ddmd.expression;
+import ddmd.func;
+import ddmd.denum;
+import ddmd.dimport;
+import ddmd.dmodule;
+import ddmd.mtype;
+import ddmd.typinf;
+import ddmd.identifier;
+import ddmd.init;
+import ddmd.doc;
+import ddmd.root.rootobject;
+import ddmd.statement;
+import ddmd.staticassert;
+import ddmd.nspace;
+import ddmd.visitor;
+
+string astTypeName(RootObject node)
+{
+    final switch(node.dyncast())
+    {
+        case DYNCAST_OBJECT:
+            return "RootObject";
+        case DYNCAST_EXPRESSION:
+            return astTypeName(cast(Expression)node);
+        case DYNCAST_DSYMBOL:
+            return astTypeName(cast(Dsymbol)node);
+        case DYNCAST_TYPE:
+            return astTypeName(cast(Type)node);
+        case DYNCAST_IDENTIFIER:
+            return "Identifier";
+        case DYNCAST_TUPLE:
+            return astTypeName(cast(Tuple)node);
+        case DYNCAST_PARAMETER:
+            return astTypeName(cast(Parameter)node);
+        case DYNCAST_STATEMENT:
+            return astTypeName(cast(Statement)node);
+    }
+}
+
+
+mixin (()
```

Comment:
I like the `()`
It allows to better distinguish function-literals from normal BlockStatements